### PR TITLE
[Modules] VM: Added proximityPlacementGroup test + changed param format

### PR DESCRIPTION
**Pull request by MrMCake**
# Description

- Added reference to proximityPlacementGroup to test files in VM & VMSS
- Changed format of diverse parameters to request resourceID instead of just name as e.g. the proximityPlacementGroup may be in a different RG or subsciption
- Replaced ProximityGroupDeployment in VMSS deployment with ProximityGroupResourceID reference to align with VM module

| Pipeline |
| - |
| [Compute: VirtualMachines](https://github.com/Azure/ResourceModules/actions/workflows/ms.compute.virtualmachines.yml) |
| [Compute: VirtualMachineScaleSets](https://github.com/Azure/ResourceModules/actions/workflows/ms.compute.virtualmachinescalesets.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

# Checklist

- [ ] I'm sure there are no other open Pull Requests for the same update/change
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (readme)
- [ ] I did format my code

### CODE REVIEW

File: .github/workflows/platform.dependencies.yml [hunk 1]
```diff
@@ -1167,7 +1167,7 @@ jobs:
             '11.azfw.parameters.json',
             '12.bastion.parameters.json',
             '13.bastion.parameters.json',
-            'parameters.json'
+            'parameters.json',
```
Comment (eriqua) [inline .github/workflows/platform.dependencies.yml:1170]:
```suggestion
            'parameters.json'
```

File: .github/workflows/platform.dependencies.yml [hunk 2]
```diff
@@ -1248,7 +1248,7 @@ jobs:
             'siterecovery.parameters.json',
             'table.parameters.json',
             'vaultcore.parameters.json',
-            'webpubsub.parameters.json'
+            'webpubsub.parameters.json',
```
Comment (eriqua) [inline .github/workflows/platform.dependencies.yml:1251]: 👻 
```suggestion
            'webpubsub.parameters.json'
```

File: .azuredevops/platformPipelines/platform.dependencies.yml [hunk 3]
```diff
@@ -52,18 +52,18 @@ stages:
...
-        - deploy_rg
-      variables:
-        templateFilePath: $(dependencyPath)/constructs/StoreVhdToStorage/deploy.bicep
-      jobs:
-        - template: /.azuredevops/pipelineTemplates/jobs.dependenciesModuleDeployment.yml
-          parameters:
-            deploymentBlocks:
-              - templateFilePath: $(templateFilePath)
-                displayName: Deploy module
+      - stage: deploy_vhd
```
Comment (eriqua) [inline .azuredevops/platformPipelines/platform.dependencies.yml:55]: 👻 👻 👻 
I'd suggest either to disable the YAML extension or to look for the right setting to disable to keep a cleaner commit history. Maybe something like yaml.editor.format = "false" or similar.

File: modules/Microsoft.Compute/virtualMachines/readme.md [hunk 4]
```diff
@@ -89,7 +89,7 @@ This module deploys one Virtual Machine with one or multiple nics and optionally
 | `pipDiagnosticSettingsName` | string | `[format('{0}-diagnosticSettings', parameters('name'))]` |  | The name of the PIP diagnostic setting, if deployed. |
 | `plan` | object | `{object}` |  | Specifies information about the marketplace image used to create the virtual machine. This element is only used for marketplace images. Before you can use a marketplace image from an API, you must enable the image for programmatic use. |
 | `provisionVMAgent` | bool | `True` |  | Indicates whether virtual machine agent should be provisioned on the virtual machine. When this property is not specified in the request body, default behavior is to set it to true. This will ensure that VM Agent is installed on the VM so that extensions can be added to the VM later. |
-| `proximityPlacementGroupName` | string | `''` |  | Resource name of a proximity placement group. |
+| `proximityPlacementGroupResourceId` | string | `''` |  | Resource name of a proximity placement group. |
```
Comment (eriqua) [inline modules/Microsoft.Compute/virtualMachines/readme.md:92]:
```suggestion
| `proximityPlacementGroupResourceId` | string | `''` |  | Resource ID of a proximity placement group. |
```

File: modules/Microsoft.Compute/virtualMachines/readme.md [hunk 5]
```diff
@@ -43,7 +43,7 @@ This module deploys one Virtual Machine with one or multiple nics and optionally
 | `additionalUnattendContent` | array | `[]` |  | Specifies additional base-64 encoded XML formatted information that can be included in the Unattend.xml file, which is used by Windows Setup. - AdditionalUnattendContent object. |
 | `adminPassword` | secureString | `''` |  | When specifying a Windows Virtual Machine, this value should be passed. |
 | `allowExtensionOperations` | bool | `True` |  | Specifies whether extension operations should be allowed on the virtual machine. This may only be set to False when no extensions are present on the virtual machine. |
-| `availabilitySetName` | string | `''` |  | Resource name of an availability set. Cannot be used in combination with availability zone nor scale set. |
+| `availabilitySetResourceId` | string | `''` |  | Resource name of an availability set. Cannot be used in combination with availability zone nor scale set. |
```
Comment (eriqua) [inline modules/Microsoft.Compute/virtualMachines/readme.md:46]:
```suggestion
| `availabilitySetResourceId` | string | `''` |  | Resource ID of an availability set. Cannot be used in combination with availability zone nor scale set. |
```

File: modules/Microsoft.Compute/virtualMachines/deploy.bicep [hunk 6]
```diff
@@ -98,10 +98,10 @@ param bootDiagnosticStorageAccountName string = ''
 param bootDiagnosticStorageAccountUri string = '.blob.${environment().suffixes.storage}/'
 
 @description('Optional. Resource name of a proximity placement group.')
```
Comment (eriqua) [inline modules/Microsoft.Compute/virtualMachines/deploy.bicep:100]:
```suggestion
@description('Optional. Resource ID of a proximity placement group.')
```

File: modules/Microsoft.Compute/virtualMachines/deploy.bicep [hunk 7]
```diff
@@ -98,10 +98,10 @@ param bootDiagnosticStorageAccountName string = ''
...
-param proximityPlacementGroupName string = ''
+param proximityPlacementGroupResourceId string = ''
 
 @description('Optional. Resource name of an availability set. Cannot be used in combination with availability zone nor scale set.')
```
Comment (eriqua) [inline modules/Microsoft.Compute/virtualMachines/deploy.bicep:103]:
```suggestion
@description('Optional. Resource ID of an availability set. Cannot be used in combination with availability zone nor scale set.')
```

File: modules/Microsoft.Compute/virtualMachines/deploy.bicep [hunk 8]
```diff
@@ -98,10 +98,10 @@ param bootDiagnosticStorageAccountName string = ''
 param bootDiagnosticStorageAccountUri string = '.blob.${environment().suffixes.storage}/'
 
 @description('Optional. Resource name of a proximity placement group.')
-param proximityPlacementGroupName string = ''
+param proximityPlacementGroupResourceId string = ''
```
Comment (eriqua) [inline modules/Microsoft.Compute/virtualMachines/deploy.bicep:101]: VMSS should also probably be aligned to the same change

File: .azuredevops/platformPipelines/platform.dependencies.yml (hunk 3 above)
Comment (MrMCake) [reply .azuredevops/platformPipelines/platform.dependencies.yml:55]: disabled it

File: modules/Microsoft.Compute/virtualMachines/deploy.bicep (hunk 8 above)
Comment (MrMCake) [reply modules/Microsoft.Compute/virtualMachines/deploy.bicep:101]: done. Removed the ProximityGroup deployment & expect resource ID instead

File: modules/Microsoft.Compute/virtualMachines/version.json [hunk 9]
```diff
@@ -1,4 +1,4 @@
 {
     "$schema": "https://raw.githubusercontent.com/dotnet/Nerdbank.GitVersioning/master/src/NerdBank.GitVersioning/version.schema.json",
-    "version": "0.4"
+    "version": "0.5"
```
Comment (eriqua) [inline modules/Microsoft.Compute/virtualMachines/version.json:3]: Let's update vmss version as well

File: .azuredevops/platformPipelines/platform.dependencies.yml (hunk 3 above)
Comment (eriqua) [reply .azuredevops/platformPipelines/platform.dependencies.yml:55]: The vhd block has one indent added. Is this intended? If not I'd suggest to rollback this change

Comment (MrMCake) [reply .azuredevops/platformPipelines/platform.dependencies.yml:55]: It's now aligned with the other `if` blocks. So I think it's actually a good automatic change :D

File: modules/Microsoft.Compute/virtualMachines/version.json (hunk 9 above)
Comment (MrMCake) [reply modules/Microsoft.Compute/virtualMachines/version.json:3]: done

File: .azuredevops/platformPipelines/platform.dependencies.yml (hunk 3 above)
Comment (eriqua) [reply .azuredevops/platformPipelines/platform.dependencies.yml:55]: Or all the others were indented by the same extension in the past 🔧

Comment (MrMCake) [reply .azuredevops/platformPipelines/platform.dependencies.yml:55]: Even if 😏😄

Comment (eriqua) [reply .azuredevops/platformPipelines/platform.dependencies.yml:55]: 🙄 😮‍💨

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**